### ISSUE TITLE
make error

### ISSUE THREAD

**Issue by wdmjun** [jrh13/hol-light#64]
File "pa_j.ml", line 413, characters 6-24:
413 |     | MtFun loc x1 x2 x3 →
            ^^^^^^^^^^^^^^^^^^
Error: The constructor MtFun expects 3 argument(s),
       but is applied here to 4 argument(s)
make: *** [Makefile:49: pa_j.cmo] Error 2

I received this error when running make in ~/hol-light.

**Comment by wdmjun** [comment]
This was on Debian Sid.

**Comment by wdmjun** [comment]
I compiled camlp5 from source. I now get the same error as #62.

**Comment by maggesi** [comment]
You need to pick up a good combination of ocaml and camlp5 versions.
The one that I'm using right now is the following:
ocaml 4.05
camlp5 7.10
there many other possibile combinations.

With opam, you can use the following instructions:

```
opam switch create 4.05.0
eval `opam env`
opam pin add camlp5 7.10
opam install num camlp5
```

**Comment by wdmjun** [comment]
Worked for me. Much thanks.

**Comment by jrh13** [comment]
Yes, thank you all! And as a result of recent work by @aqjune-aws, I think getting working combinations is now much less problematic generally; most recent versions will work and if all else fails you can install HOL Light itself from opam.
